containers/ruby/.devcontainer/library-scripts/ruby-debian.sh

```diff
@@ -247,25 +247,48 @@ git clone --depth=1 \
     -c fetch.fsck.zeroPaddedFilemode=ignore \
     -c receive.fsck.zeroPaddedFilemode=ignore \
     https://github.com/rbenv/rbenv.git /usr/local/share/rbenv
-ln -s /usr/local/share/rbenv/bin/rbenv /usr/local/bin
-updaterc 'eval "$(rbenv init -)"'
+
 git clone --depth=1 \
     -c core.eol=lf \
     -c core.autocrlf=false \
     -c fsck.zeroPaddedFilemode=ignore \
     -c fetch.fsck.zeroPaddedFilemode=ignore \
     -c receive.fsck.zeroPaddedFilemode=ignore \
     https://github.com/rbenv/ruby-build.git /usr/local/share/ruby-build
+
 mkdir -p /root/.rbenv/plugins
-ln -s /usr/local/share/ruby-build /root/.rbenv/plugins/ruby-build
+
 if [ "${USERNAME}" != "root" ]; then
     mkdir -p /home/${USERNAME}/.rbenv/plugins
-    chown -R ${USERNAME} /home/${USERNAME}/.rbenv
-    ln -s /usr/local/share/ruby-build /home/${USERNAME}/.rbenv/plugins/ruby-build
 fi
 
+# Update bashrc and zshrc with conditionals
+# We use "not false" so that the default state is true, i.e., enabled
+updaterc 'if [ "${RVM_ENABLED}" != "false" ]; then 
+    source /usr/local/rvm/scripts/rvm 
+fi'
+
+# All commands have the output redirected to /dev/null so that there is no output on shell startup.
+if [ "${USERNAME}" != "root" ]; then
+    mkdir -p /home/${USERNAME}/.rbenv/plugins
+    updaterc 'if [ "${RBENV_ENABLED}" != "false" ]; then
+        chown -R ${USERNAME} /home/${USERNAME}/.rbenv
+        ln -s /usr/local/share/ruby-build /home/${USERNAME}/.rbenv/plugins/ruby-build > /dev/null 2>&1
+    else
+        rm -f /home/${USERNAME}/.rbenv/plugins/ruby-build > /dev/null 2>&1
+    fi'
+fi
+
+updaterc 'if [ "${RBENV_ENABLED}" != "false" ]; then
+        sudo ln -s /usr/local/share/rbenv/bin/rbenv /usr/local/bin > /dev/null 2>&1
```

Comment:
This will not work in the majority of images since `sudo` is typically not available unless explicitly installed.  I don't know that there's really a problem with it being in the path in either the rvm or rbenv case is there?  

If neither are sourced I wouldn't expect to see conflicts. Or is there something I'm not aware of?

If being in the path is a problem, perhaps a better approach is to reference the full path in the next line instead and remove the symlink entirely.

```
eval "$(/usr/local/share/rbenv/bin/rbenv init -)"
```

You could also explicitly update the PATH in this scenario if that's needed (export PATH=$PATH:/usr/local/share/rbenv/bin)

Reply:
There is not an explicit problem, I think the only reason we did it this way is to try to keep the user path as clean as possible. However, it's not actually necessary, so we could remove it.

Reply:
I do like the idea of just updating the PATH directly though, so we'll do that.